share/picongpu/tests/CurrentDeposition/lib/python/test/CurrentDeposition/MainTest.py

```diff
@@ -0,0 +1,205 @@
+    """creates a series of the data found at path.
+    path should have the form data_%T to provide access to all iterations.
+
+    returns: "./simData_%T.h5"
+        series
+        order... order of assignement function
+        parameter list... list of parameters of the simulation (units etc.)"""
+    # load series
+    series = io.Series(path, io.Access.read_only)
+
+    if len(series.iterations) == 1:
+        print("There is just 1 iteration in the series")
+        print("make sure, there are at least two")
+        longer_series = False
+
+    else:
+        longer_series = True
+
+    # read order assignment function
+    i = series.iterations[0]
+    electrons = i.particles["e"]
+    order = int(electrons.get_attribute("particleShape"))
+
+    # read parameters of the simulation
+    cell_depth = i.get_attribute("cell_depth")
+    cell_height = i.get_attribute("cell_height")
+    cell_width = i.get_attribute("cell_width")
+    dt = i.get_attribute("dt")
+    unit_time = i.get_attribute("unit_time")
+    unit_length = i.get_attribute("unit_length")
+    unit_charge = i.get_attribute("unit_charge")
+
+    return longer_series, series, order, [
+        cell_depth, cell_height, cell_width, dt, unit_time, unit_charge,
+        unit_length
+        ]
+
+
+def read_series(series):
+    """reads the position values position_offset (grid_knot) and position
+    (float in the intervall [0,1), relativ position in the cell) and the
+    current density provided by the simulation as well the charge of the
+    particle.
+
+    returns:
+        Js... current density list (contains array of each component for each
+                                    iteration)
+        poss... array of all relative in-Cell positions
+        pos_offs... array of all knot-positions
+        charge.. charge of particle
+        """
+    num_iter = len(series.iterations)
+
+    # create arrays for saving the data
+    Js = []
+    poss = np.zeros((num_iter, 3))
+    pos_offs = np.zeros((num_iter, 3))
+
+    for iteration in range(0, num_iter, 1):
+        i = series.iterations[iteration]  # current iteration
+
+        # current density field
+        J_x = i.meshes["J"]["x"].load_chunk()
+        J_y = i.meshes["J"]["y"].load_chunk()
+        J_z = i.meshes["J"]["z"].load_chunk()
+
+        # charge
+        charge = i.particles["e"]["charge"][io.Mesh_Record_Component.SCALAR]
+
+        # node coordinates
+        pos2_x = i.particles["e"]["position"]["x"].load_chunk()
+        pos2_y = i.particles["e"]["position"]["y"].load_chunk()
+        pos2_z = i.particles["e"]["position"]["z"].load_chunk()
+
+        # InCell coordinates
+        pos_off2_x = i.particles["e"]["positionOffset"]["x"].load_chunk()
+        pos_off2_y = i.particles["e"]["positionOffset"]["y"].load_chunk()
+        pos_off2_z = i.particles["e"]["positionOffset"]["z"].load_chunk()
+
+        series.flush()
+
+        # write coordinatetupel
+        pos2 = np.array([*pos2_x, *pos2_y, *pos2_z])
```

Comment:
This unpacking and repacking via a preceding aster ix as about 3x less efficient as writing the arraies directly. (I did some timing with ipythons `%timeit`.) Please use
```suggestion
        pos2 = np.array([pos2_x, pos2_y, pos2_z])
```
instead.

Reply:
Okay factor 3x is for a 3x5 array, it reaches a factor >100x for a 3x1M array. 
So please change.

Reply:
ok, thanks for the advise.

Reply:
I have to unpack the data anyways. I tested the *-version against pos2[1][0] and this took ca. 2x longer for the pos2[1][0] version. So I will stick with the *

Reply:
I still do net get this yet.
If I execute:
```python
import numpy as np

a = np.array([1.2], dtype=np.float32)
b = np.array([3.4], dtype=np.float32)
c = np.array([5.6], dtype=np.float32)

%%timeit
x = np.array([*a, *b, *c])

# vs

%%timeit
x = np.array([a, b, c])
```

the later one is always faster (both on x86_64 (hemera) and ARM (M1)).

hemera: 3.31 µs ± 0.10 µs vs 1.6 µs ± 0.17 µs.

@MRZirper what do you mean with "pos2[1][0]"?

@steindev I could not reproduce the speedup you have seen on a list/array of 1 entry. We need to discuss what is going on here.

Reply:
@MRZirper please provide your test code as a snippet here.

Reply:
If we chose the second  purely method one, we would calculated the coordinates and therefore the current Deposition with a 1x1 array instead of an number. This caused some issues so that the test fails.

We would instead call something like coordinate = x[1][0] to get an number. If we include this in the speedtest (what I have done): the *-Version outperforms the normal-array-metode in most runs.

Code:
`import numpy as np
import timeit

a = [1]
b = [1]
c = [1]

start = timeit.default_timer()
for i in range(10000):
    d = np.array([*a,*b,*c])
    out1 = d[0]
    out2 = d[1]
    out3 = d[2]
stop = timeit.default_timer()
print(f"Time with a * = {stop - start}s")

start = timeit.default_timer()
for i in range(10000):
    d = np.array([a,b,c])
    out1 = d[0][0]
    out2 = d[1][0]
    out3 = d[2][0]
stop = timeit.default_timer()
print(f"Time without = {stop - start}s")`

Reply:
The two versions from the original suggestion by @PrometheusPi do different things:
```python3
>>> np.array([*pos2_x, *pos2_y, *pos2_z]).shape
(3, )
>>> np.array([pos2_x, pos2_y, pos2_z])
(3, 1)
```
That seems to be where the confusion comes from. Haven't run the test but is efficiency really a concern here?